### ISSUE TITLE
[3][sschiessl-bcp] Reconnect to socket on failure

### ISSUE THREAD

**Issue by wmbutler** [bitshares/bitshares-ui#1485]
**Describe the bug**
When the browser has been disconnected from the socket, but it is still open, the wallet ceases to work. It throws this message with every click:

```
Error in MarketsActions.unSubscribeMarket:  Error: websocket state error:3
    at e.call (app.js:43)
    at e.exec (app.js:45)
    at app.js:29
    at Object.f [as unSubscribeMarket] (app.js:50)
    at t.value (3.js:19975)
    at f.unmountComponent (app.js:45)
    at Object.unmountComponent (app.js:23)
    at f.unmountComponent (app.js:45)
    at Object.unmountComponent (app.js:23)
    at f.unmountComponent (app.js:45)
```

**To Reproduce**
1. Open the wallet
2. Disconnect from the Internet
3. Wait a few minutes and reconnect to the Internet
4. Click around and monitor your console log

**Expected behavior**
When this error is detected, reconnect to the socket.

**Screenshots**

**Desktop (please complete the following information):**

 - Chrome, latest OS X

**Comment by sschiessl-bcp** [comment]
I assume you had the exchange tab open there?

#1407 automatically reconnects, I want to include active tab recognition as well.

**Comment by wmbutler** [comment]
I'll break this down more comprehensively.

Steps to reproduce.

1. Load the wallet.bitshares.org
1. Confirm node is connnect in lower right
1. Open inspector (network)
1. Disable WIFI and Ethernet connection
1. Observe that it takes about 60 seconds for DISCONNECTED state in lower right corner
1. Turn WIFI or Ethernet back on
1. Observe that browser never reconnects to socket
1. Refresh the browser
1. Socket reconnects

In a perfect world, the browser should:

1. Notice that the socket is gone within 15 seconds and make it clear to the user that it's not connected
1. Reconnect to the socket gracefully soon after the network is restored.

![disconnect](https://user-images.githubusercontent.com/1254810/42126111-0cf0d2a4-7c48-11e8-9092-9c7425a6fa62.gif)

**Comment by sschiessl-bcp** [comment]
This is included in the PR for #1407

**Comment by sschiessl-bcp** [comment]
@wmbutler please include in 180721 invoice
